src/compiler/utilities.ts

```diff
@@ -2560,10 +2560,8 @@ namespace ts {
         return false;
     }
 
-    export function isStringOrNumericLiteral(node: Node): node is StringLiteral | NumericLiteral {
-        const kind = node.kind;
-        return kind === SyntaxKind.StringLiteral
-            || kind === SyntaxKind.NumericLiteral;
+    export function isStringOrNumericLiteralLike(node: Node): node is StringLiteralLike | NumericLiteral {

```

Comment:
The name implies that this includes BigInt literals once they are added to the language. I don't know if that is correct for all call sites

Reply:
It looks like in most cases it would be -- `{ [0n]: true }` is the same as `{ 0: true }` so as long as IntLiteral#text doesn't include `n` this would work. @sheetalkamat is working on bigints.

Reply:
@andy-ms #25886 I think is the nearly complete implementation at this point (unless we have multiple implementations at the ready). But yeah, I guess indexing with `bigint` is fine - it's just odd, since in the typesystem that needs to go under the `number` index signature since `0n` and `0` have the same string representation. Probably worth testing in #25886, at least! It's probably not a concern for this PR directly, since it'll need to be dealt with at the point they're added.